lib/middleware.js

```diff
@@ -0,0 +1,151 @@
+'use strict'
+
+const fs = require('fs')
+const tmp = require('tmp')
+const execa = require('execa')
+const debug = require('debug')('pdf-service')
+
+const catchCmdlineError = (err, res) => {
+  const stderr = err.stderr
+  const newlineIndex = stderr.indexOf('\n')
+  debug(stderr.substr(0, newlineIndex))
+  res.status(500).send(err)
+}
+
+module.exports = {
+  setHeaders (req, res, next) {
+    debug('setting headers')
+    res.set('Access-Control-Allow-Origin', '*')
+    res.set('Access-Control-Allow-Headers', 'Origin, X-Requested-With, Content-Type, Accept')
+    res.set('Content-Type', 'application/pdf')
+    next()
+  },
+
+  setupTemp (req, res, next) {
+    debug('setup temp storage')
+    req.temp = tmp.dirSync({unsafeCleanup: true})
+    next()
+  },
+
+  writeHtmlFile (req, res, next) {
+    debug('write html file')
+    fs.writeFile(`${req.temp.name}/html.html`, req.body.html, err => {
+      if (err) res.status(500).send(err)
```

Comment:
you probably shouldn't be handling errors in your middleware. You should call next(err) and let error handling middleware do the job of catching all errors, generating html user friendly errors